src/azure-cli/azure/cli/command_modules/role/_params.py

```diff
@@ -202,10 +202,21 @@ def load_arguments(self, _):
         c.argument('end_time', help=time_help.format('end time', 'the current time'))
 
     with self.argument_context('role assignment create') as c:
-        PrincipalType = self.get_models('PrincipalType', resource_type=ResourceType.MGMT_AUTHORIZATION)
-        if PrincipalType:
-            c.argument('assignee_principal_type', min_api='2018-09-01-preview', arg_type=get_enum_type(PrincipalType),
-                       help='use with --assignee-object-id to avoid errors caused by propagation latency in AAD Graph')
+        # PrincipalType = self.get_models('PrincipalType', resource_type=ResourceType.MGMT_AUTHORIZATION)
+
+        # A temporary fix for https://github.com/Azure/azure-cli/issues/11594.
+        # As only `User', 'Group' or 'ServicePrincipal' are allowed values, the REST spec contains invalid values
+        # (like MSI) which are used only internally by the service. So hide them.
+        # https://github.com/Azure/azure-rest-api-specs/blob/962013a1cf9bf5b87e3aad75a14c7dd620acda62/specification/authorization/resource-manager/Microsoft.Authorization/preview/2020-04-01-preview/authorization-RoleAssignmentsCalls.json#L508-L522
+        from enum import Enum
+
+        class PrincipalType(str, Enum):
+            user = "User"
+            group = "Group"
+            service_principal = "ServicePrincipal"
+
+        c.argument('assignee_principal_type', min_api='2018-09-01-preview', arg_type=get_enum_type(PrincipalType),
```

Comment:
I remember that there  are several other allowed values and there may be some users who are using them. Are you sure all these values are invalid so that we can remove them directly ?

Reply:
I am sure. According to the error message from service:

> The PrincipalType property 'MSI' is not valid. It must be 'User', 'Group' or 'ServicePrincipal'.

```
> az role assignment create --scope /subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/resourceGroups/rg1111 --role "Reader" --assignee-principal-type MSI --assignee-object-id 2fdcd421-c744-440d-b253-3daf68c697eb --debug

msrest.http_logger : Request URL: 'https://management.azure.com/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/resourceGroups/rg1111/providers/Microsoft.Authorization/roleAssignments/0a5f880e-f02f-4a3e-8799-0a09fd95cf29?api-version=2020-04-01-preview'
msrest.http_logger : Request method: 'PUT'
msrest.http_logger : Request body:
msrest.http_logger : {"properties": {"roleDefinitionId": "/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/providers/Microsoft.Authorization/roleDefinitions/acdd72a7-3385-48ef-bd42-f606fba81ae7", "principalId": "2fdcd421-c744-440d-b253-3daf68c697eb", "principalType": "MSI"}}

msrest.http_logger : Response status: 400
msrest.http_logger : Response content:
msrest.http_logger : {"error":{"code":"InvalidPrincipalType","message":"The PrincipalType property 'MSI' is not valid. It must be 'User', 'Group' or 'ServicePrincipal'."}}
```

See email: _BadRequestError: The PrincipalType property 'MSI' is not valid. It must be 'User', 'Group' or 'ServicePrincipal'_

Reply:
`arg_type=get_enum_type(['User', 'Group', 'ServicePrincipal'])` for short

reference: [registering enums](https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#registering-enums)

Reply:
I prefer to leave it as it is now, as I am only patching `PrincipalType` to make as less code change as possible. `PrincipalType` will be fetched from Python SDK again once the REST spec is fixed.